aws/ec2metadata/api.go

```diff
@@ -20,6 +20,20 @@ func (c *EC2Metadata) GetMetadata(p string) (string, error) {
 	return output.Content, req.Send()
 }
 
+// GetDynamicData uses the path provided to request
+func (c *EC2Metadata) GetDynamicData(p string) (string, error) {
+	op := &request.Operation{
+		Name:       "GetMetadata",
+		HTTPMethod: "GET",
+		HTTPPath:   path.Join("/", "dynamic", p),
+	}
+
+	output := &metadataOutput{}
+	req := c.NewRequest(op, nil, output)
+
+	return output.Content, req.Send()
```

Comment:
What do you think about calling `GetMetadata` instead of generating the request directly? Maybe similar to `Region`?

Its counter to the generated service clients, but in this case it simplifies the code and is possible since these requests to metadata similar.